File: Components/WinForms/CS/GaugesMainDemo/Modules/StateIndicatorGaugeFeatures.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.Skins;
using DevExpress.XtraGauges.Core.Drawing;
using DevExpress.XtraGauges.Core.Model;
using DevExpress.XtraGauges.Win.Base;
using DevExpress.XtraGauges.Win.Gauges.Circular;

namespace DevExpress.XtraGauges.Demos {
    public partial class StateIndicatorGaugeFeatures : TutorialControl {
        public StateIndicatorGaugeFeatures() {
            InitializeComponent();
        }
        protected override void Dispose(bool disposing) {
            if(disposing && (components != null)) {
                components.Dispose();
            }
            base.Dispose(disposing);
        }
        #region StateIndicator Gauge
        void StateIndicatorGaugeFeatures_Load(object sender, EventArgs e) {
            smilesState.Value = stateIndicatorComponent3.StateIndex;
            arrowsState.Value = stateIndicatorComponent2.StateIndex;
            elLightsState.Value = stateIndicatorComponent1.StateIndex;
            trafficLightState.Value = stateIndicatorComponent4.StateIndex;

            this.trafficLightState.EditValueChanged += new System.EventHandler(this.OnTrafficLightStateChanged);
            this.elLightsState.EditValueChanged += new System.EventHandler(OnElLightStateChanged);
            this.arrowsState.EditValueChanged += new System.EventHandler(OnArrowStateChanged);
            this.smilesState.EditValueChanged += new System.EventHandler(OnSmileStateChanged);
            this.currencyState.EditValueChanged += new System.EventHandler(OnCurrencyStateChanged);
            this.flagsState.EditValueChanged += new System.EventHandler(OnFlagStateChanged);
        }
        void OnFlagStateChanged(object sender, EventArgs e) {
            stateIndicatorComponent6.StateIndex = flagsState.Value;
        }
        void OnCurrencyStateChanged(object sender, EventArgs e) {
            stateIndicatorComponent5.StateIndex = currencyState.Value;
        }
        void OnTrafficLightStateChanged(object sender, EventArgs e) {
            stateIndicatorComponent4.StateIndex = trafficLightState.Value;
        }
        void OnSmileStateChanged(object sender, EventArgs e) {
            stateIndicatorComponent3.StateIndex = smilesState.Value;
        }
        void OnArrowStateChanged(object sender, EventArgs e) {
            stateIndicatorComponent2.StateIndex = arrowsState.Value;
        }
        void OnElLightStateChanged(object sender, EventArgs e) {
            stateIndicatorComponent1.StateIndex = elLightsState.Value;
        }
        #endregion
    }
}

File: Components/WinForms/VB/GaugesMainDemo/Modules/StateIndicatorGaugeFeatures.vb
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.Skins
Imports DevExpress.XtraGauges.Core.Drawing
Imports DevExpress.XtraGauges.Core.Model
Imports DevExpress.XtraGauges.Win.Base
Imports DevExpress.XtraGauges.Win.Gauges.Circular

Namespace DevExpress.XtraGauges.Demos
    Partial Public Class StateIndicatorGaugeFeatures
        Inherits TutorialControl

        Public Sub New()
            InitializeComponent()
        End Sub
        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            If disposing AndAlso (components IsNot Nothing) Then
                components.Dispose()
            End If
            MyBase.Dispose(disposing)
        End Sub
        #Region "StateIndicator Gauge"
        Private Sub StateIndicatorGaugeFeatures_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            smilesState.Value = stateIndicatorComponent3.StateIndex
            arrowsState.Value = stateIndicatorComponent2.StateIndex
            elLightsState.Value = stateIndicatorComponent1.StateIndex
            trafficLightState.Value = stateIndicatorComponent4.StateIndex

            AddHandler trafficLightState.EditValueChanged, AddressOf OnTrafficLightStateChanged
            AddHandler elLightsState.EditValueChanged, AddressOf OnElLightStateChanged
            AddHandler arrowsState.EditValueChanged, AddressOf OnArrowStateChanged
            AddHandler smilesState.EditValueChanged, AddressOf OnSmileStateChanged
            AddHandler currencyState.EditValueChanged, AddressOf OnCurrencyStateChanged
            AddHandler flagsState.EditValueChanged, AddressOf OnFlagStateChanged
        End Sub
        Private Sub OnFlagStateChanged(ByVal sender As Object, ByVal e As EventArgs)
            stateIndicatorComponent6.StateIndex = flagsState.Value
        End Sub
        Private Sub OnCurrencyStateChanged(ByVal sender As Object, ByVal e As EventArgs)
            stateIndicatorComponent5.StateIndex = currencyState.Value
        End Sub
        Private Sub OnTrafficLightStateChanged(ByVal sender As Object, ByVal e As EventArgs)
            stateIndicatorComponent4.StateIndex = trafficLightState.Value
        End Sub
        Private Sub OnSmileStateChanged(ByVal sender As Object, ByVal e As EventArgs)
            stateIndicatorComponent3.StateIndex = smilesState.Value
        End Sub
        Private Sub OnArrowStateChanged(ByVal sender As Object, ByVal e As EventArgs)
            stateIndicatorComponent2.StateIndex = arrowsState.Value
        End Sub
        Private Sub OnElLightStateChanged(ByVal sender As Object, ByVal e As EventArgs)
            stateIndicatorComponent1.StateIndex = elLightsState.Value
        End Sub
        #End Region
    End Class
End Namespace
